test/model/internal_link_test.dart

```diff
@@ -130,13 +130,16 @@ void main() {
       final bool expected = testCase.$1;
       final String description = testCase.$2;
       final String urlString = testCase.$3;
+      final String urlWithChannelSyntax = urlString.replaceFirst('#narrow/stream/', '#narrow/channel/');
```

Comment:
nit: Feels like this can go unnoticed when skimming through tests, instead I'd just add those extra cases in the `testCases` list.